tests/lax_control_flow_test.py

```diff
@@ -2050,29 +2106,54 @@ def sqrt_cubed(x, tangent_solve=scalar_solve):
     self.assertAllClose(
         results, 5.0**1.5, check_dtypes=False, rtol={np.float64: 1e-7})
 
+  @parameterized.named_parameters([
+    {"testcase_name": "vector_solve", "solve_method": vector_solve},
+    {"testcase_name": "newton_raphson", "solve_method": newton_raphson},
+  ])
   @jtu.skip_on_flag("jax_skip_slow_tests", True)
-  def test_custom_root_vector_with_solve_closure(self):
-
-    def vector_solve(f, y):
-      return jnp.linalg.solve(jax.jacobian(f)(y), y)
+  def test_custom_root_vector_with_solve_closure(self, solve_method):
 
     def linear_solve(a, b):
       f = lambda y: high_precision_dot(a, y) - b
       x0 = jnp.zeros_like(b)
       solution = jnp.linalg.solve(a, b)
       oracle = lambda func, x0: solution
-      return lax.custom_root(f, x0, oracle, vector_solve)
+      return lax.custom_root(f, x0, oracle, solve_method)
 
     rng = np.random.RandomState(0)
     a = rng.randn(2, 2)
     b = rng.randn(2)
     jtu.check_grads(linear_solve, (a, b), order=2,
-                    atol={np.float32: 1e-2, np.float64: 1e-11})
+                    atol={np.float32: 1e-2, np.float64: 1e-11},
+                    modes=["fwd"],
+    )
 
     actual = jax.jit(linear_solve)(a, b)
     expected = jnp.linalg.solve(a, b)
     self.assertAllClose(expected, actual)
 
+  def test_custom_root_vector_nonlinear(self):
+
+    def nonlinear_func(x, y):
+      # func(x, y) == 0 if and only if x == y.
+      return (x - y) * (x**2 + y**2 + 1)
+
+    def tangent_solve(g, y):
+      return linear_equation_solve(jax.jacobian(g)(y), y)
+
+    def nonlinear_solve(y):
+      f = lambda x: nonlinear_func(x, y)
+      x0 = -jnp.ones_like(y)
+      return lax.custom_root(f, x0, newton_raphson, tangent_solve)
+
+    rng = np.random.RandomState(0)
+    y = rng.randn(3, 1)
```

Comment:
Note: If we use a scalar instead of an array (e.g., `y = 3.0`), reverse mode works in this test. See comment above.